src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx

```diff
@@ -66,42 +94,61 @@ const PaginatedSelect: FunctionComponent<Props> = ({
               )}
             </Dropdown>
           </IntersectionProvider>
-        </ClickOutside>
-      )}
-    >
-      {({ toggleVisibility, ref, visible }) => (
-        <Button
-          className={cn(styles.button, className)}
-          variant="flat"
-          adornmentLeft
-          color="mono"
-          onClick={toggleVisibility}
-          ref={ref}
-          uppercase={false}
-        >
-          {Icon && (
-            <ButtonSvgIcon className={styles.buttonIconLeft} Icon={Icon} />
-          )}
-          <Flex alignItems="center" className={styles.wrapper}>
-            {selected}
+        )}
+      >
+        {({ toggleVisibility, ref }) => (
+          <Flex
+            className={cn(styles.button, className)}
+            onClick={handleButtonClick}
+            ref={ref}
+            justifyContent="space-between"
+          >
+            {Icon && (
+              <ButtonSvgIcon className={styles.buttonIconLeft} Icon={Icon} />
+            )}
+            {open && !!onFilter ? (
+              <Localized id="admin-paginatedSelect-filter">
+                <TextField
+                  onChange={(e) => onFilter(e.target.value)}
+                  ref={filterRef}
+                  aria-roledescription="Filter results"
+                />
+              </Localized>
+            ) : (
+              <Flex
+                alignItems="center"
+                className={styles.wrapper}
+                style={{
+                  // display: open && !!onFilter ? "none" : "inherit",
+                  textOverflow: "ellipsis",
+                  whiteSpace: "nowrap",
+                }}
+                role="button"
+                tabIndex={0}
+                onFocus={() => setOpen(true)}
+                onBlur={() => setOpen(false)}
+              >
+                {selected}
```

Comment:
In the dropdown, in the design, `All sites` has a separator beneath it, but I'm not seeing that here. Should the styles be updated to include this separator?

Reply:
✅

Reply:
thanks for adding! It looks like it has more spacing than the design, and maybe the spacing should be more like 2? Can you take a look?

Reply:
Agreed, 1 actually looked closer to me. Lemme know if you disagree!